include/hipSYCL/sycl/libkernel/generic/hiplike/builtins.hpp

```diff
@@ -216,6 +216,14 @@ HIPSYCL_HIPLIKE_BUILTIN double __hipsycl_rsqrt(double x) noexcept {
 
 HIPSYCL_DEFINE_HIPLIKE_MATH_BUILTIN(__hipsycl_sin, sinf, sin)
 
+template<class T, class FloatPtr>
+HIPSYCL_HIPLIKE_BUILTIN T __hipsycl_sincos(T x, FloatPtr cosval) noexcept {
+  *cosval = hiplike_builtins::__hipsycl_cos(x);
+  return hiplike_builtins::__hipsycl_sin(x);
```

Comment:
Shouldn't this return `sin(*cosval)` instead of `sin(x)`?

Reply:
Nevermind - spec says it should return `sin(x)` *and* `cos(x)`, not `sin(cos(x))` as I had assumed.